spec/models/guest_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe Guest, type: :model do
+  describe 'relationships' do
+    it { should have_many :guest_rooms }
+    it { should have_many(:rooms).through(:guest_rooms) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of :name }
+  end
+
+  describe 'class methods' do
+    before do 
+      Guest.create!(name: "Letti")
+      Guest.create!(name: "Calum")
+      Guest.create!(name: "Nika")
+    end
+
+    it "::total_count" do
+      expect(Guest.total_count).to eq(3)
+
+      Guest.create!(name: "Melony")
+
+      expect(Guest.total_count).to eq(4)
+    end
+
+    it "::uniq_guest_names" do
+      expect(Guest.uniq_guest_names).to eq(["Nika", "Letti", "Calum"])
+
+      Guest.create!(name: "Nika")
+
+      expect(Guest.uniq_guest_names).to eq(["Nika", "Letti", "Calum"])
+
+      Guest.create!(name: "Melony")
+
+      expect(Guest.uniq_guest_names).to eq(["Nika", "Letti", "Calum", "Melony"])
```

Comment:
Similar here to above; the extension asks for this list to be all the guests _that stayed at this Hotel_. I think a more straightforward way of implementing this would be to make an instance method on Hotel that gets all of its related `guests`, maybe through an association.... Or, you could write a join starting in Hotel and ending up in Guests. 

Also, this test doesn't really cover "uniqueness" since the guests are just created and then listed. The "uniqueness" of each Guest in this scenario really comes from a relationship between Hotel and Guest, through the GuestRoom join table. In other words, even if a Guest stayed a Hotel 2-3-4 times, they should only appear in this list _once_.